### PR TITLE
Search: Record Meter Design Updates

### LINKED ISSUE #24102: Search: address feedback from Record Meter design review
@jeffgolenski provided us with some excellent feedback on the Record Meter in pcNPJE-Nn-p2. The adjustments needed are listed below.

- [x] Notice box: fix spacing/padding
- [ ] Notice box: use info icon from WordPress Design Library / Gutenberg
- [x] Notice box: all copy and icons should be black (or red for an important notice)
- [x] Notice box: Rounded border should be `gray 5` from our color library
- [x] Notice box: improve error message for 'unable to index your content'
- [x] Meter: fix spacing and alignment (see diagram below)
- [x] Meter: can we make the meter 24px tall?
- [x] Meter: on mobile - reduce vertical space between multiple lines in legend (to 24px)

image

### PR DESCRIPTION
**Pull request by annacmc**
This PR addresses all the feedback from @jeffgolenski  in pcNPJE-Nn#comment-1197 with exception to the error notice box changes (the icon used, and the multi-line notice display wording/spacing)

Fixes #24102

#### Changes proposed in this Pull Request:
* Notice Boxes: All warning notice text and icons are now black
* Notice Boxes: All warning notice boxes rounded border is now gray 5 from our color library
* Notice Boxes: All error notice boxes text, icon & borders are red from our color library
* Content spacing: The space between the header and “Your search records” title is now 64px
* Font size: The text below “Your search records” as well as the post types and warning messages are now all the same (1em)
* Record Meter: Chart height is now 24px tall
* Record Meter alignment: The meter is now flush with the text content on the left.
* Record limit surpassed warning:  Double checked this has been removed
* Metadata: Confirmed this is removed
* The vertical space between the post types when they’re on multiple lines is now 24px on mobile (and also not on mobile, as the entire chart legend was tightened up) 

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Go to a site with an active Jetpack Search subscription
* Go to /wp-admin/admin.php?page=jetpack-search&features=record-meter
* Check the record meter still works & looks correctly, and that the above mentioned additions are displaying correctly. 
* 
| Record Meter  | With Warning Notice | With Error Notice |
| ------------- | ------------- | ------------- |
| image | image | image |

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/style.scss [hunk 1]
```diff
@@ -39,17 +42,12 @@ ul.jp-search-chart-legend {
 .jp-search-notice-box {
 	background-color: white;
 	color: black;
-	border: lightgray 0.5px solid;
+	border: #dcdcde 0.5px solid; // grey #5
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/style.scss:45]: You should be able to import these colours as Sass variables - like this:

```
@import "~@automattic/color-studio/dist/color-variables";

button {
  background: $studio-pink-50;
  color: $studio-white;
}
```

(from https://github.com/Automattic/color-studio/blob/master/README.md)

Comment (annacmc) [reply projects/packages/search/src/dashboard/components/record-meter/style.scss:45]: Ah, thank you so much! I was wondering about this, so asked over in the P2 post but didn't think to ask it in here :) That will help a lot.

**Review by jsnmoon — DISMISSED** [review]
Code changes look good! Deferring to Jeff for another round of design feedback.

**Comment by bluefuton** [comment]
👍 I think Jeff is currently at a meetup so I suggest we push on and merge this, then have another round of design feedback later.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**